### PR TITLE
Allow plugins to modify routeInfo

### PR DESCRIPTION
**Pull request by andreme**
## Description

Allow plugins to modify routeInfo to for example add or modify the data that is used on a page.

## Changes/Tasks

- Call new routeInfo hook in exportRoute
- Call new routeInfo hook in dev server for each request
- Extract valid hooks from object, to remove extra list

## Motivation and Context

#1566

## Types of changes

- [ ] Refactoring/add tests (refactoring or adding test which isn't a fix or add a feature)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have updated the documentation accordingly
- [x] I have updated the CHANGELOG with a summary of my changes
- [ ] My changes have tests around them

### CODE REVIEW

File: packages/react-static/src/static/plugins.js [hunk 1]
```diff
@@ -1,39 +1,6 @@
 import { getHooks, reduceHooks } from '../utils'
 
-const supportedHooks = [
```
Comment (SleeplessByte) [inline packages/react-static/src/static/plugins.js:3]: What's the reason these ae removed? Even though they don't have a default implementation, are you certain they're not actually supported anymore?

Comment (andreme) [reply packages/react-static/src/static/plugins.js:3]: All the entries in the array are also keys in the exported object. So I just removed the duplication. validatePlugin now takes the valid hook names from the exported object.

Comment (SleeplessByte) [reply packages/react-static/src/static/plugins.js:3]: GitHub was collapsing those lines, so I didn't realise. This is good. Thanks :)

**Review by SleeplessByte — APPROVED** [review]

**Comment by SleeplessByte** [comment]
(It's still a draft, but overall this is looking good to me)

**Comment by andreme** [comment]
Thanks @SleeplessByte

**Pull request merged**